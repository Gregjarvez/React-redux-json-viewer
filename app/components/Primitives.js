import React from 'react';
import { string, shape } from 'prop-types';
import Copy from 'react-icons/lib/md/note-add';

const Primitive = props => (
  <tr className="layout--row">
    <td className="model--panel">
      <span className="model--copy" title="copy path to key">
        <Copy />
      </span>
    </td>
    <td className="model--key">
      <table style={{ marginLeft: props.meta.margin }}>
        <tbody>
          <tr className="layout--row">
            <td className="model--key">{ props.Qey }</td>
            <td>:</td>
            <td className={`model--value ${props.meta.type === 'string'
              ? 'model--value-string'
              : 'model--value-number'}`}
            >{ props.value }
            </td>
          </tr>
        </tbody>
      </table>
    </td>
  </tr>
);


Primitive.propTypes = {
  Qey: string.isRequired,
  value: string.isRequired,
  meta: shape({
    type: string,
    margin: string,
  }).isRequired,

};
export default Primitive;
