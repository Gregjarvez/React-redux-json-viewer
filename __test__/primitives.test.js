import React from 'react';
import { shallow } from 'enzyme';
import Primitives from '../app/components/Primitives';


describe('<Primitives />', () => {
  it('renders correctly', () => {
    const component = shallow(<Primitives
      Qey="String"
      value="Hello"
      meta={{}}
    />);
    expect(component).toMatchSnapshot();
  });


  it('contains 5 table cell in all', () => {
    const component = shallow(<Primitives
      Qey="String"
      value="Hello"
      meta={{}}
    />);
    expect(component.find('td').length).toEqual(5);
  });
});
