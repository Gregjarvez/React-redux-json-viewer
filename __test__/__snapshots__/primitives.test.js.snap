// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Primitives /> renders correctly 1`] = `
<tr
  className="layout--row"
>
  <td
    className="model--panel"
  >
    <span
      className="model--copy"
      title="copy path to key"
    >
      <MdNoteAdd
        onClick={[Function]}
      />
    </span>
  </td>
  <td
    className="model--key"
  >
    <table
      style={
        Object {
          "marginLeft": "undefinedpx",
        }
      }
    >
      <tbody>
        <tr
          className="layout--row"
        >
          <td
            className="model--key"
          >
            String
          </td>
          <td>
            :
          </td>
          <td
            className="model--value model--value-number"
          >
            Hello
          </td>
        </tr>
      </tbody>
    </table>
  </td>
</tr>
`;
